File: scenes/settlements/Road.cs
using System;
using Godot;
using Godot.Collections;
using Array = System.Array;

namespace nuclearnation
{
    public class Road:Reference
    {
        private readonly Vector2 from;
        private readonly Vector2 to;

        public bool IsVertical()
        {
            return (int)from.x == (int)to.x;
        }

        public Road(Vector2 from, Vector2 to)
        {
            this.from = from;
            this.to = to;
        }

        public Array<Vector2> GetTiles()
        {
            var to_x = to.x;
            if ((int)to_x == (int)from.x)
            {
                to_x += 1;
            }
            
            var to_y = to.y;
            if ((int)to_y == (int)from.y)
            {
                to_y += 1;
            }
            
            var response = new Array<Vector2>();
            for (var x = @from.x; x < to_x; x++)
            {
                for (var y = @from.y; y < to_y; y++)
                {
                    response.Add(new Vector2(x,y));
                }
            }

            return response;
        }


    }
}

File: FS/Road.fs
module FS.Road

open Godot

type Road(from:Vector2,``to``:Vector2) =
    inherit Reference()
    
    member val from:Vector2 = from
    member val ``to``:Vector2 = ``to``
    
    member this.IsVertical() =
        int from.x = int ``to``.x
    
    member this.GetTiles():List<Vector2> =
        let to_x:int =
            let tmp_to_x = int (if ((int)``to``.x = (int)from.x) then int ``to``.x + 1 else int ``to``.x)
            if (this.IsVertical() = false) then tmp_to_x else tmp_to_x - 1
        let to_y:int =
            let tmp_to_y = int (if ((int)``to``.y = (int)from.y) then int ``to``.y + 1 else int ``to``.y)
            if (this.IsVertical()) then tmp_to_y else tmp_to_y - 1
        
        let xs = seq{for i in int from.x .. to_x do yield i}
        let ys = seq{for i in int from.y .. to_y do yield i}
        
        let vectors = seq {
            for x in xs do
                for y in ys do yield Vector2(float32 x,float32 y)
        }
        
        Seq.toList vectors
       
    
